core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -79,6 +82,13 @@ private[spark] abstract class Task[T](
       metrics)
     TaskContext.setTaskContext(context)
     taskThread = Thread.currentThread()
+
+    val callerContext =
+      s"Spark_AppId_${appId.getOrElse("")}_AppAttemptId_${appAttemptId.getOrElse("None")}" +
+        s"_JobId_${jobId.getOrElse("0")}_StageID_${stageId}_stageAttemptId_${stageAttemptId}" +
+        s"_taskID_${taskAttemptId}_attemptNumber_${attemptNumber}"
+    Utils.setCallerContext(callerContext)
```

Comment:
we should move this below the if(_killed) check as not reason to log this if it doesn't run.